llvm/lib/Transforms/Yk/LivenessAnalysis.cpp

```diff
@@ -83,8 +83,27 @@ LivenessAnalysis::LivenessAnalysis(Function *Func) {
       //
       // The book doesn't cover this quirk, as it explains liveness for
       // non-SSA form, and thus doesn't need to worry about Phi nodes.
-      if (isa<PHINode>(I))
+      if (isa<PHINode>(I)) {
+        PHINode *P = cast<PHINode>(&I);
+        for (unsigned IVC = 0; IVC < P->getNumIncomingValues(); IVC++) {
+          BasicBlock *IBB = P->getIncomingBlock(IVC);
+          Value *IV = P->getIncomingValue(IVC);
+          if (isa<Constant>(IV)) {
+            continue;
+          }
+          // For each block that isn't the incoming block create a Def for this
+          // value. This means when we do the backwards liveness pass this
+          // value is immediately killed in the block.
+          for (auto *BBB = P->block_begin(); BBB != P->block_end(); BBB++) {
+            if (*BBB != IBB) {
+              Instruction *First = &((*BBB)->back());
```

Comment:
Do I see it right that you are getting the first, not the last instruction? Rename the variable?

Reply:
[6326cd3](https://github.com/ykjit/ykllvm/pull/41/commits/6326cd30a3e63b74215560523f8cb3854c067feb)